### PR TITLE
AuthRuleWrapper patch

### PR DESCRIPTION
**Pull request by ki1729**
# Description

Adding default contstructor to AuthRulesWrapper

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescription.java [hunk 1]
```diff
@@ -141,6 +141,11 @@ private static final class AuthorizationRulesWrapper {
             namespace = "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect")
         private final List<AuthorizationRuleImpl> items;
 
+        @JsonCreator
+        private AuthorizationRulesWrapper() {
+            this.items = new ArrayList<AuthorizationRuleImpl>();
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescription.java:146]: Could this be replaced with `Collections.emptyList()` so we don't initialise a new array?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescription.java [hunk 2]
```diff
@@ -141,6 +141,11 @@ private static final class AuthorizationRulesWrapper {
             namespace = "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect")
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescription.java:141]: Missing CHANGELOG entry

Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/models/QueueDescription.java:141]: Added :)

**Review by conniey — APPROVED** [review]
:shipit:

**Pull request merged**